src/Microsoft.AspNetCore.Blazor.Browser/Interop/JavaScriptInvoke.cs

```diff
@@ -0,0 +1,132 @@
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Blazor.Browser.Interop
+{
+    internal class JavaScriptInvoke
+    {
+        public static string InvokeDotnetMethod(string methodOptions, string methodArguments)
+        {
+            // We invoke the dotnet method and wrap either the result or the exception produced by
+            // an error into an invocation result type. This invocation result is just a discriminated
+            // union with either success or failure.
+            try
+            {
+                return InvocationResult<object>.Success(InvokeDotNetMethodCore(methodOptions, methodArguments));
+            }
+            catch (Exception e)
+            {
+                var exception = e;
+                while (exception.InnerException != null)
+                {
+                    exception = exception.InnerException;
+                }
```

Comment:
This isn't super right. What are you really after? If you're trying to get rid of `TargetInvocationException` then just handle that.

Reply:
Most likely, I don't remember the exact pattern when you are using tasks instead of async/await, so I just did something quick.